caddyconfig/caddyfile/parse.go

```diff
@@ -348,10 +359,20 @@ func (p *parser) doImport() error {
 		}
 	}
 
+	// copy the tokens so we don't overwrite p.definedSnippets
+	tokensCopy := make([]Token, len(importedTokens))
+	copy(tokensCopy, importedTokens)
```

Comment:
These two lines can be replaced by just this:
```suggestion
	tokensCopy := importedTokens[:]
```

Reply:
@Mohammed90 They are different, though -- your suggestion does not perform a copy, so any changes to the elements of the slice will be reflected in both importedTokens and tokensCopy.

Reply:
Nope, it won't reflect on the original. I've tested on the playground.
https://play.golang.org/p/P2MjWaDexhL

It's `tokensCopy := importedTokens` vs `tokensCopy := importedTokens[:]` that determines whether it's value copy or pointer copy.

Reply:
Am I missing something slice internals pitfall?

Reply:
I see another one-liner here citing more benefits: https://github.com/go101/go101/wiki/How-to-perfectly-clone-a-slice%3F

Reply:
@Mohammed90 Your example is wrong, I think, to demonstrate a copy. Here is the corrected version: https://play.golang.org/p/EVDWDqA9ntL (see line 12)

Reply:
I just tried the above @Mohammed90, doesn't work. Both log filenames end up being `a.example.com`.

Reply:
Ah I see. It goes in one direction not the other.

Reply:
This does work though, but looks really ugly IMO.

```go
tokensCopy := append(importedTokens[:0:0], importedTokens...)
```

Reply:
@Mohammed90 The `[:]` notation creates a new slice _header_ over an underlying array (often used for converting arrays to slices, or for copying a header of an existing slice -- but the underlying elements stay the same, so if you change `a[0]` or `b[0]` it changes it for both.

Reply:
I've just played with the example more and realized what I missed 🤦‍♂️